pandas/core/indexes/category.py

```diff
@@ -422,6 +422,17 @@ def where(self, cond, other=None):
         cat = Categorical(values, dtype=self.dtype)
         return type(self)._simple_new(cat, name=self.name)
 
+    def putmask(self, mask, value):
```

Comment:
not necessarily for today. but is there any value to pushing this down to the array and having a putmask_compat until NEP18 can be supported?

putmask on the Index returns a copy whereas putmask compat on the array would be expected to be inplace. This may not be so easy for Categorical, but for other numpy backed arrays could be more trivial.

also is the goal of extension array backed indexes to allow 3rd party EAs in the Index. If so, putmask on the array would need to be added to the EA interface?

Reply:
> [...] putmask on the array would need to be added to the EA interface?

I would be in favor of this

Reply:
yep I am also +1 on this as this is a 'standard' array method, can you create an issue (we might have one?)